src/box/sql/select.c

```diff
@@ -740,8 +740,10 @@ addWhereTerm(Parse * pParse,	/* Parsing context */
 
 	struct Expr *pE1 = sql_expr_new_column(db, pSrc, iLeft, iColLeft);
 	struct Expr *pE2 = sql_expr_new_column(db, pSrc, iRight, iColRight);
-	if (pE1 == NULL || pE2 == NULL)
+	if (pE1 == NULL || pE2 == NULL) {
```

Comment:
AFAIU if `pE1` allocation succeeds but `pE2` allocations fails, `pE1` will leak. This is hardly possible in reality, but static analyzers may still complain about it.

Reply:
True, fixed.